src/services/review/ReviewRequestService.ts

```diff
@@ -196,13 +215,103 @@ export default class ReviewRequestService {
           createdAt: new Date(created_at).getTime(),
           // TODO!
           newComments: 0,
-          // TODO!
-          firstView: false,
+          firstView: isFirstView,
         }
       })
     )
   }
 
+  markAllReviewRequestsAsViewed = async (
+    sessionData: UserWithSiteSessionData,
+    site: Site
+  ): Promise<void> => {
+    const { isomerUserId: userId } = sessionData
+
+    const requestsViewed = await this.reviewRequestView.findAll({
+      where: {
+        siteId: site.id,
+        userId,
+      },
+    })
+
+    const allActiveRequests = await this.repository.findAll({
+      where: {
+        siteId: site.id,
+        // NOTE: Closed and merged review requests would not have an
+        // entry in the review request views table
+        reviewStatus: ["OPEN", "APPROVED"],
+      },
+    })
+
+    const requestIdsViewed = requestsViewed.map(
+      (request) => request.reviewRequestId
+    )
+    const allActiveRequestIds = allActiveRequests.map((request) => request.id)
+    const requestIdsToMarkAsViewed = _.difference(
+      allActiveRequestIds,
+      requestIdsViewed
+    )
+
+    requestIdsToMarkAsViewed.forEach(async (requestId) => {
```

Comment:
this might actually be blocking sequentially (see [here](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)). even though semantically, `forEach` is more indicative of a side effect, we might wanna `Promise.all` and `map` here to ensure greater throughput (if it matters. if this endpoint is relatively fast/we have a reasonable bound for array size, fel free to ignore)

Reply:
Good point! I forgot about this, adjusted in 5ff10c4.